tiledb/sm/c_api/tiledb.cc

```diff
@@ -2907,6 +2961,60 @@ int32_t tiledb_query_get_buffer_var_nullable(
   return TILEDB_OK;
 }
 
+int32_t tiledb_query_get_data_buffer(
+    tiledb_ctx_t* ctx,
+    tiledb_query_t* query,
+    const char* name,
+    void** buffer,
+    uint64_t** buffer_size) {
+  // Sanity check
+  if (sanity_check(ctx) == TILEDB_ERR || sanity_check(ctx, query) == TILEDB_ERR)
+    return TILEDB_ERR;
+
+  // Set attribute buffer
```

Comment:
I believe this should be `get` rather than `set`. Same with the following 2 APIs.